lib/resque/failure/redis.rb

```diff
@@ -48,14 +48,18 @@ def self.all(offset = 0, limit = 1, queue = nil)
       end
 
       def self.each(offset = 0, limit = self.count, queue = :failed, class_name = nil, order = 'desc')
+        if class_name
+          original_limit = limit
+          limit = count
+        end
         all_items = limit == 1 ? [all(offset,limit,queue)] : Array(all(offset, limit, queue))
         reversed = false
         if order.eql? 'desc'
           all_items.reverse!
           reversed = true
         end
         all_items.each_with_index do |item, i|
-          if !class_name || (item['payload'] && item['payload']['class'] == class_name)
+          if !class_name || (item['payload'] && item['payload']['class'] == class_name && (original_limit -= 1) >= 0)
```

Comment:
This would be more readable if the added condition was `&& original_limit > 0` and the decrement was moved inside the if block as just `original_limit -= 1`.  It will also be more readable with the `original_limit` -> `limit` rename as suggested above